tests/classes/core/PKPComponentRouterTest.php

```diff
@@ -271,6 +271,7 @@ public function testRoute()
         // with a reference to the request object as the first argument.
         Registry::set('request', $this->request);
         $user = new \PKP\user\User();
+        $user->setData('id', 0);
```

Comment:
Could you add a comment documenting why this is required? It's probably something that'll cause confusion later.

Reply:
Added the following comment 

```
/*
* Set the id of the user here to something other than null in order for the UserRolesRequiredPolicy 
* to be able to work as it supposed to.
* Specifically, the UserRolesRequiredPolicy::effect calls the getByUserIdGroupedByContext function
* which needs a userId that is not nullable.
*/
```